pvanalytics/quality/irradiance.py

```diff
@@ -1 +1,333 @@
 """Quality control functions for irradiance data."""
+
+import numpy as np
+from pvlib.tools import cosd
+
+
+QCRAD_LIMITS = {'ghi_ub': {'mult': 1.5, 'exp': 1.2, 'min': 100},
+                'dhi_ub': {'mult': 0.95, 'exp': 1.2, 'min': 50},
+                'dni_ub': {'mult': 1.0, 'exp': 0.0, 'min': 0},
+                'ghi_lb': -4, 'dhi_lb': -4, 'dni_lb': -4}
+
+QCRAD_CONSISTENCY = {
+    'ghi_ratio': {
+        'low_zenith': {
+            'zenith_bounds': [0.0, 75],
+            'ghi_bounds': [50, np.Inf],
+            'ratio_bounds': [0.92, 1.08]},
+        'high_zenith': {
+            'zenith_bounds': [75, 93],
+            'ghi_bounds': [50, np.Inf],
+            'ratio_bounds': [0.85, 1.15]}},
+    'dhi_ratio': {
+        'low_zenith': {
+            'zenith_bounds': [0.0, 75],
+            'ghi_bounds': [50, np.Inf],
+            'ratio_bounds': [0.0, 1.05]},
+        'high_zenith': {
+            'zenith_bounds': [75, 93],
+            'ghi_bounds': [50, np.Inf],
+            'ratio_bounds': [0.0, 1.10]}}}
+
+
+def _check_limits(val, lb=None, ub=None, lb_ge=False, ub_le=False):
+    """Return True where lb < (or <=) val < (or <=) ub."""
+    if lb_ge:
+        lb_op = np.greater_equal
+    else:
+        lb_op = np.greater
+    if ub_le:
+        ub_op = np.less_equal
+    else:
+        ub_op = np.less
+
+    if (lb is not None) & (ub is not None):
+        return lb_op(val, lb) & ub_op(val, ub)
+    elif lb is not None:
+        return lb_op(val, lb)
+    elif ub is not None:
+        return ub_op(val, ub)
+    else:
+        raise ValueError('must provide either upper or lower bound')
+
+
+def _qcrad_ub(dni_extra, sza, lim):
+    cosd_sza = cosd(sza)
+    cosd_sza[cosd_sza < 0] = 0
+    return lim['mult'] * dni_extra * cosd_sza**lim['exp'] + lim['min']
+
+
+def check_ghi_limits_qcrad(ghi, solar_zenith, dni_extra, limits=None):
```

Comment:
Is it better style to do:
```python
def check_ghi_limits_qcrad(ghi, solar_zenith, dni_extra, limits=QCRAD_LIMITS):
```
and remove the 
```python
    if not limits:
        limits = QCRAD_LIMITS
```
block below?

Reply:
The same pattern is used in `check_[dni/dhi]_limits

Reply:
The second pattern is better, since `dict` is mutable, it's discouraged to use as a default value (could be changed inside the function and the change persists outside).